halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/config/v1/HalconfigParser.java

```diff
@@ -74,41 +74,49 @@ Halconfig parseConfig(InputStream is) throws UnrecognizedPropertyException {
     }
   }
 
+  public InputStream getHalconfigStream() throws FileNotFoundException {
+    return new FileInputStream(new File(halconfigPath));
+  }
+
   /**
    * Returns the current halconfig stored at the halconfigPath.
    *
    * @see Halconfig
    * @param reload if we should check the disk for the halconfig.
    * @return the fully parsed halconfig.
    */
-  public Halconfig getConfig(boolean reload) {
+  public Halconfig getHalconfig(boolean reload) {
+    Halconfig res = null;
+
     if (!reload && halconfig != null) {
-      halconfig.parentify();
-      return halconfig;
+      res = halconfig;
+    } else {
+      try {
+        InputStream is = getHalconfigStream();
+        res = parseHalconfig(is);
+      } catch (FileNotFoundException e) {
+      } catch (UnrecognizedPropertyException e) {
```

Comment:
The `ParseConfigExceptions` dispatch constructors based on the exception type - and the `Exception2 | Exception2` squashes the exception type to their common super-class.